library/ecp.c

```diff
@@ -2666,14 +2666,17 @@ static int mbedtls_ecp_mul_shortcuts( mbedtls_ecp_group *grp,
 
     if( mbedtls_mpi_cmp_int( m, 0 ) == 0 )
     {
+        MBEDTLS_MPI_CHK( mbedtls_ecp_check_pubkey( grp, P ) );
```

Comment:
question: seems a shame to repeat these checks - was it not possible to put this at the top and remove the check in `ecp_mul_restartable_internal()`?

Reply:
No, this function is (only) called from `mbedtls_ecp_muladd_restartable()`, which says that it validates its two ECP point arguments to check they are on the curve. So putting the checks here ensures that `mbedtls_ecp_muladd_restartable()` does what it says in all cases.

`mbedtls_ecp_mul_restartable()` also says that it validates its argument, which it does currently do in all cases, so that check needs to stay.